x-pack/plugins/transform/public/app/common/validators/is_json_string.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Validates whether string input can be parsed as a valid JSON
+ * @param value User input value.
+ */
+export function isJsonString(value: unknown): boolean {
+  if (typeof value !== 'string') return false;
+
+  try {
+    return !!JSON.parse(value);
```

Comment:
`JSON.parse(true)` returns `true` as well. So you should also check for type `string`.

Reply:
We already have a check in there to only allows string. I added another jest test to check for `true` here [5321409](https://github.com/elastic/kibana/pull/173736/commits/532140900e590d0163e1be5704c4bddb62d8f363).